dashboard/test/lib/datastore_cache_test.rb

```diff
@@ -3,110 +3,160 @@
 require 'dynamic_config/adapters/memory_adapter'
 
 class DatastoreCacheTest < ActiveSupport::TestCase
-  class FakeListener
-    attr_reader :changed
-
-    def initialize
-      reset
-    end
-
-    def reset
-      @changed = false
-    end
-
-    def on_change
-      @changed = true
-    end
-  end
-
   def setup
     @data_adapter = MemoryAdapter.new
+    @datastore_cache_identifier = 'default datastore cache for unit tests'
+    @datastore_cache = DatastoreCache.new(@data_adapter, @datastore_cache_identifier)
   end
 
   test 'simple get and set' do
-    cache = DatastoreCache.new @data_adapter
-    key = "key"
-    value = [1, 3, 2]
+    key = 'key'
+    first_value = [1, 3, 2]
+    second_value = [2, 4, 3]
 
-    cache.set(key, value)
-    assert_equal value, cache.get(key)
-  end
+    assert_nil @datastore_cache.get(key)
 
-  test 'simple get and set with multiple listeners' do
-    cache = DatastoreCache.new @data_adapter
-
-    listener = FakeListener.new
-    cache.add_change_listener(listener)
-
-    key = "key"
-    value = [1, 3, 2]
-
-    refute listener.changed, 'on_change should not be called before change'
-    cache.set(key, value)
-    assert_equal value, cache.get(key)
-    assert listener.changed, 'on_change should be called after local set'
-    listener.reset
-
-    # Add another listener, and verify that it receives change notifications too.
-    value2 = [2, 4, 3]
-    listener2 = FakeListener.new
-    cache.add_change_listener(listener2)
-    cache.set(key, value2)
-    assert_equal value2, cache.get(key)
-    assert listener.changed
-    assert listener2.changed
-    listener.reset
-    listener2.reset
-
-    # Set the same value and verify that the change listeners are not called.
-    cache.set(key, value2)
-    refute listener.changed
-    refute listener2.changed
-  end
+    @datastore_cache.set(key, first_value)
+    assert_equal first_value, @datastore_cache.get(key)
 
-  test 'cache is refreshed after expiration time' do
-    key = "key"
-    value = "1, 2, 3"
+    @datastore_cache.set(key, second_value)
+    assert_equal second_value, @datastore_cache.get(key)
+  end
 
-    @data_adapter.set(key, 'not the real value')
-    cache = DatastoreCache.new @data_adapter, cache_expiration: 0.01
+  test 'local_cache_expired? is updated on refresh and after time passes' do
+    cache = DatastoreCache.new @data_adapter, 'fast expiration', cache_expiration: 0.01
+    # Cache starts out fresh
+    refute cache.local_cache_expired?
 
-    listener = FakeListener.new
-    cache.add_change_listener(listener)
+    # Cache expires after a configurable time period has elapsed
+    sleep 0.1
```

Comment:
Could we use the Rails [travel_to](https://api.rubyonrails.org/v6.1.4.7/classes/ActiveSupport/Testing/TimeHelpers.html#method-i-travel_to) test helpr instead of `sleep`?

Reply:
Oh, yeah, that looks much better; I was just thoughtlessly following the pattern established by existing tests. I think that's an improvement that makes sense to incorporate in this PR

Reply:
Neat! I hadn't seen `travel_to` before, great feature.